src/controllers/comment.js

```diff
@@ -0,0 +1,115 @@
+import models from '../db/models';
+import { errorResponse, commentResponse, getCommentsAuthors } from '../helpers';
+
+const { Comment: CommentModel } = models;
+
+/**
+ * @description A controller class for handling comment business logic
+ *
+ * @class Comment
+ */
+class Comment {
+  /**
+   * @description - Gets all comments associated with an article
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - containing an array of comments
+   *
+   * @memberof Comment
+   */
+  static async getAll(request, response) {
+    const { article } = request;
+    try {
+      const comments = await article.getComments({
+        attributes: ['id', 'createdAt', 'userId', 'updatedAt', 'body']
+      });
+
+      getCommentsAuthors(comments)
+        .then(authors => response.send(commentResponse(comments, authors)))
+        .catch(error => response.send(errorResponse([error.message])));
+    } catch (error) {
+      response.status(500).send(errorResponse([error.message]));
+    }
+  }
+
+  /**
+   * @description - Gets an article comment by it's ID
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - a single comment
+   *
+   * @memberof Comment
+   */
+  static async getOne(request, response) {
+    const { article, params } = request;
+    const { commentId } = params;
+
+    try {
+      const comments = await article.getComments({
+        attributes: ['id', 'createdAt', 'userId', 'updatedAt', 'body'],
+        where: { id: parseInt(commentId, 10) }
+      });
+
+      if (!comments.length) return response.status(404).send(errorResponse([`Comment with id ${commentId} not found`]));
+
+      getCommentsAuthors(comments)
+        .then((authors) => {
```

Comment:
Hi Luqman, I understand what you are trying to do here, but I think it could write fewer lines of codes by awaiting the result of 'getCommentsAuthors(comments),' and then perform an action on it.

Reply:
Oh! Good catch @frostyblok 🥇. 
It's fixed now. Thanks for taking the time to review. 👍

Reply:
OK good job, Luqman!